quil-rs/benches/simplification.rs

```diff
@@ -0,0 +1,26 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use once_cell::sync::Lazy;
+use quil_rs::expression::Expression;
+use std::str::FromStr;
+
+static EXPRESSIONS: Lazy<Vec<Expression>> = Lazy::new(|| {
+    include_str!("expressions.txt")
+        .lines()
+        .map(|line| Expression::from_str(line).expect("these are valid expressions"))
+        .collect()
+});
+
+fn simplify(e: Expression) -> Expression {
+    e.into_simplified()
+}
+
+fn benchmark_expression_simplification(c: &mut Criterion) {
+    EXPRESSIONS.iter().enumerate().for_each(|(i, e)| {
+        c.bench_function(&format!("expression_{i}"), |b| {
+            b.iter(|| black_box(simplify(e.clone())))
+        });
+    })
```

Comment:
if we rely on the vec index to key the inputs, that means we can't remove, change, or reorder them without tampering with our results. maybe a prefix on each line with the index, then lines can be removed without consequence?